epoch.go

```diff
@@ -2832,27 +2987,35 @@ func (e *Epoch) processEmptyNotarization(emptyNotarization *EmptyNotarization) e
 	return e.processReplicationState()
 }
 
-func (e *Epoch) processLatestRoundReceived(latestRound *QuorumRound, from NodeID) error {
+// processQuorumRound processes a quorum round received from another node.
+// It verifies the quorum round and stores it in the replication state if valid.
+func (e *Epoch) processQuorumRound(latestRound *QuorumRound, from NodeID) error {
```

Comment:
Not related to this PR, but thought I should mention it.

If we replicate a notarized block $b_i$ that was only notarized by the node that sent us the block and all other nodes notarized the empty round, the next notarized block we receive after that $b_{i+1}$, can be a block building on top of the parent of $b_i$. 

Conversely, we can replicate an empty notarization while everyone else but the sender node has a regular notarization for the same round. Then we might get stuck, won't we?